### ISSUE TITLE
Not returning null for queries involving nullable columns

### ISSUE THREAD

**Issue by jordangignac** [typeorm/typeorm#408]
If I am to create an entity with a nullable column and attempt to query the data the nullable column is not returned at all unless it is not `null`. I would assume that it would return `null`, simply omitting the data can sometimes be confusing, and could possibly cause issues with a legacy client of ours expecting the values. A simplified example follows...

**TestEntity.ts**
```javascript
@Entity()
export class Test {
    @PrimaryGeneratedColumn()
    id: number;

    @Column({ nullable: true })
    tested: string;
}
```

**Main.ts**
```javascript
const newTest = new Test();
connection.entityManager.persist(newTest);
```

**Result**
```json
{ "id": 1 }
```

**Expected**
```json
{ "id": 1, "tested": null }
```

My question would be is this the intended behaviour? Am I missing something or a way to query the data including any nullable columns?

**Comment by valVk** [comment]
Asked the same question in the chat. This is confusing for front end developers because they have to check whether field is defined AND not null or empty object, etc.

**Comment by pleerock** [comment]
I don't think persist method should change something beside `id`. Persist method also performs partial updates, and if you want to partially update lets say `title` `persist({ id: 1, title: "xxx" })` and since its partial updation it means original object already can have name, etc. other properties defined in the database, it means if persist method will return then `{ id: 1, title: "xxx", name: null }` it won't be correct.
 
Different story for the queries that return models (using `find*` and `QueryBuilder`) - shall they return nulls for empty fields or not. Something that needs more thoughts.

**Comment by valVk** [comment]
> Different story for the queries that return models (using find* and QueryBuilder) - shall they return nulls for empty fields or not.

in all ORM's with which I've worked, when fields has `null` then they will return `{ id: 1234, name: null }` after query will be executed. In that case even if this field will not be set up, developer always will know that this field exists.

Here is example of Stripe's [documentation](https://stripe.com/docs/api#create_charge)
![selection_20170424_b9d289f](https://cloud.githubusercontent.com/assets/1130626/25341271/f630122c-2910-11e7-81a0-f8ba06838fba.png)
'application' attribute is not documented but I know that in some cases it may be presented in a response.

In my case, I'm doing response to the request of front end application and because of insufficient of the fields, I and front end developer must search a work around, in my case I'm thinking about `AfterLoad` event

**Comment by jordangignac** [comment]
@pleerock That was my mistake when creating the issue, the 'Result' and 'Expected' JSON objects were returned using the `find` query after the object had been persisted.

I had not thought about it and can completely understand the persist method returning partial objects for the case of partial updates. But I do believe for queries that return models, like you said above, it would be beneficial to return `null` values like @valVk mentioned to maintain a more 'transparent' api and consistency.

**Comment by pleerock** [comment]
Okay now when you fetch columns with nulls they are returned to you. Available from `0.1.0-alpha.21`
